### ISSUE TITLE
Do we consider epoch or stochastic gradient step as a unit for local steps?

### ISSUE THREAD

**Issue by omarfoq** [owkin/FLamby#58]
In the current implementation `num_updates` (in `strategies\fed_avg.py`) corresponds to the number of stochastic gradient updates at each round. This is correct as most known convergence results are provided when `num_updates` is constant across clients, however I have previously seen papers, including the first FL papers, considering a `num_epochs` as unit for local steps.  Do we want to keep the current definition for `num_updates`, or do we want to give the possibility to choose between the two options?
